tests/framework/statistics/core.py

```diff
@@ -0,0 +1,96 @@
+# Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+"""Consumer of statistics."""
+
+
+from datetime import datetime
+from collections import namedtuple, defaultdict
+from typing_extensions import TypedDict
+from framework.statistics.producer import Producer
+from framework.statistics.consumer import Consumer
+
+
+# pylint: disable=R0903
+class Statistics(TypedDict):
+    """Data class for aggregated statistic results."""
+
+    name: str
+    iterations: int
+    results: dict
+    custom: dict
+
+
+Pipe = namedtuple("Pipe", "producer consumer")
+
+
+class Core:
+    """Base class for statistics core driver."""
+
+    # pylint: disable=W0102
+    def __init__(self, name, iterations, custom={}):
+        """Core constructor."""
+        self._pipes = defaultdict(Pipe)
+        self._statistics = Statistics(name=name,
+                                      iterations=iterations,
+                                      results={},
+                                      custom=custom)
+
+    def add_pipe(self, producer: Producer, consumer: Consumer, tag=None):
```

Comment:
Are there use-cases for more pipes ? Do we need this ?

Reply:
Oh, I see. This might explain the overall complexity of the statistics component. I have to give it more thought.